.github/workflows/check_docs.sh

```diff
@@ -5,6 +5,6 @@ KATANA_REPO_ROOT="$(cd "$(dirname "$0")/../.."; pwd)"
 BUILD_DIR=$1
 shift 1
 
-cmake -S . -B $BUILD_DIR \
-  -DCMAKE_TOOLCHAIN_FILE=$BUILD_DIR/conan_paths.cmake && \
-  cd $BUILD_DIR && $KATANA_REPO_ROOT/scripts/check_docs.sh
+cmake -S . -B "$BUILD_DIR" \
+  -DCMAKE_TOOLCHAIN_FILE="$BUILD_DIR"/conan_paths.cmake && \
```

Comment:
I prefer to quote whole arguments instead of just the variables being expanded. So unless someone else objects I would prefer `"-DCMAKE_TOOLCHAIN_FILE=$BUILD_DIR/conan_paths.cmake"`. I think it makes the argument boundaries clearer.

This comment applies throughout the PR. I can mark all the spot if you need, but I suspect you can find them easily enough.